frontend/pages/queries/details/components/QueryReport/QueryReport.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState, useContext, useEffect } from "react";
+// import { Row, Column } from "react-table";
+
+// import classnames from "classnames";
+import FileSaver from "file-saver";
+import { QueryContext } from "context/query";
+
+import {
+  generateCSVFilename,
+  generateCSVQueryResults,
+} from "utilities/generate_csv";
+import { IQueryReport } from "interfaces/query_report";
+import { humanLastSeen } from "utilities/helpers";
+
+import Button from "components/buttons/Button";
+import Icon from "components/Icon/Icon";
+import TableContainer from "components/TableContainer";
+import ShowQueryModal from "components/modals/ShowQueryModal";
+
+import generateResultsTableHeaders from "./QueryReportTableConfig";
+
+interface IQueryReportProps {
+  queryReport?: IQueryReport;
+}
+
+const baseClass = "query-report";
+const CSV_TITLE = "Query";
+
+const tableResults = (results: any) => {
+  return results.map((result: any) => {
+    const hostInfo = {
+      host_display_name: result.host_name,
+      last_fetched: result.last_fetched,
+    };
+
+    const tableData = { ...hostInfo, ...result.columns };
```

Comment:
can we make this clearer by changing the names:
`tableData` –> `allRowColumns`
`hostInfo` –> `hostInfoColumns`
and maybe add a comment to the effect of "result.columns holds the variable columns that will differ per query, while hostInfo represents the host metadata that is returned with every query."